docs/source/developer_guide/library_design.md

```diff
@@ -0,0 +1,111 @@
+# Library Design
+
+cuSpatial has two main components,
+the cuSpatial-python package and libcuspatial c++ package.
+This article dicusses the design of cuspatial-python,
+for information of libcuspatial,
+see [libcuspatial developer guide](link) and [c++ API reference](link).
+We refer to the python package as `cuspatial` and the c++ package as `libcuspatial` below.
+
+## Overview
+
+At a high level, `cuspatial` has three parts:
+- A GPU backed `GeoDataFrame` data structure
+- A set of computation APIs
+- A cython API layer
+
+## GPU Accelerated `GeoDataFrame` and `GeoSeries`
+
+Note: since the core data structure of cuspatial shares the same name as that
+of `geopandas`.
+We refer to geopandas' dataframe object as `geopandas.GeoDataFrame` and to
+cuspatial's dataframe object as `GeoDataFrame`.
+
+Under the hood,
+cuspatial can perform parallel computation on geometry data thanks to its [structure of array](https://en.wikipedia.org/wiki/Parallel_array) (SoA) format.
+Specifically,
+cuspatial adopts geoarrow format as the SoA standard.
+Geoarrow is a derived data type from the arrow list type adopting a [`Variable-size List Layout`](https://arrow.apache.org/docs/format/Columnar.html#variable-size-list-layout),
+with the inner-most layer storing the points with a `Fixed-size list layout` array with `size=2`.
+Per definition,
+each increase in geometry complexity (dimension, or multi-geometry) requires an extra level of indirection.
+Geoarrow allows a mixture of geometry types to present in the same column by adopting the [Dense Union Array Layout](https://arrow.apache.org/docs/format/Columnar.html#dense-union).
+In cusptial,
+we refer to each level of indirection from highest level to lowest by
+`geometries`, `parts`, `rings` and `coordinates`.
+The first three are integral offset arrays and the last is an floating-point interleaved xy-coordinated array.
+Read [geoarrow format documentation](https://github.com/geopandas/geo-arrow-spec/blob/main/format.md) specification for more detail.
```

Comment:
```suggestion
----------------------------------------------------------------------------------------------------
Under the hood, cuspatial can perform parallel computation on geometry
data thanks to its
[structure of arrays](https://en.wikipedia.org/wiki/Parallel_array) (SoA)
format. Specifically, cuspatial adopts geoarrow format. Geoarrow is derived
from the Apache Arrow list type, and it adopts a
[`Variable-size List Layout`](https://arrow.apache.org/docs/format/Columnar.html#variable-size-list-layout),
with the inner-most layer storing the points in a `Fixed-size list layout` array
with `size==2`. 

By definition, each increase in geometry complexity (dimension, or multi-
geometry) requires an extra level of indirection. Geoarrow allows a mixture
of geometry types to be present in the same column by adopting the
[Dense Union Array Layout](https://arrow.apache.org/docs/format/Columnar.html#dense-union).
In cuSpatial, we use the following names for the levels of indirection from
highest level to lowest: `geometries`, `parts`, `rings` and `coordinates`. The
first three are integral offset arrays and the last is a floating-point
interleaved xy-coordinate array.

Read the [geoarrow format specification](https://github.com/geopandas/geo-arrow-spec/blob/main/format.md)
for more detail.
```

Reply:
After the second read, I think the second paragraph on quick overview of the geoarrow features can be broken down into two parts:

```suggestion
----------------------------------------------------------------------------------------------------
Under the hood, cuspatial can perform parallel computation on geometry
data thanks to its
[structure of arrays](https://en.wikipedia.org/wiki/Parallel_array) (SoA)
format. Specifically, cuspatial adopts geoarrow format. Geoarrow is derived
from the Apache Arrow list type, and it adopts a
[`Variable-size List Layout`](https://arrow.apache.org/docs/format/Columnar.html#variable-size-list-layout),
with the inner-most layer storing the points in a `Fixed-size list layout` array
with `size==2`. 

By definition, each increase in geometry complexity (dimension, or multi-
geometry) requires an extra level of indirection. In cuSpatial, we use the following names for the levels of indirection from
highest level to lowest: `geometries`, `parts`, `rings` and `coordinates`. The
first three are integral offset arrays and the last is a floating-point
interleaved xy-coordinate array.

Geoarrow also allows a mixture
of geometry types to be present in the same column by adopting the
[Dense Union Array Layout](https://arrow.apache.org/docs/format/Columnar.html#dense-union).

Read the [geoarrow format specification](https://github.com/geopandas/geo-arrow-spec/blob/main/format.md)
for more detail.
```